src/main/java/ogc/rs/apiserver/handlers/DxTokenAuthenticationHandler.java

```diff
@@ -68,11 +72,20 @@ public DxTokenAuthenticationHandler(Vertx vertx, JsonObject config) {
                   .setAudience(Collections.singletonList(config.getString("audience")));
               jwtAuthOptions.getJWTOptions().setIssuer(config.getString("issuer"));
               jwtAuth = JWTAuth.create(vertx, jwtAuthOptions);
+
+              // Mark as successfully initialized
+                jwtAuthInitialized = true;
+                initializationError = null;
+                LOGGER.info("JWT Authentication successfully initialized");
             })
         .onFailure(
             handler -> {
               LOGGER.error("failed to get JWT public key from auth server");
               LOGGER.error("Authentication verticle deployment failed.");
+                // Mark as failed initialization
+                jwtAuthInitialized = false;
```

Comment:
See if it's also possible to throw a FATAL logger here and a RuntimeException. Check what happens when the latter is thrown

Reply:
Also see if vertx.close() does something